lightningd/plugin.c

```diff
@@ -781,9 +792,9 @@ static void plugin_manifest_cb(const char *buffer,
 	const jsmntok_t *resulttok;
 
 	/* Check if all plugins have replied to getmanifest, and break
-	 * if they are */
+	 * if they have and this is the startup init */
 	plugin->plugins->pending_manifests--;
-	if (plugin->plugins->pending_manifests == 0)
+	if (plugin->plugins->pending_manifests == 0 && !getenv("LIGHTNINGD_PLUGIN"))
```

Comment:
Abusing the env var to see if we're starting up is pretty horrible!

Better to have a 'startup' bool, or simply set pending_manifests to (size_t)-1 every time you start a plugin after startup, so this never happens?

Reply:
> Abusing the env var to see if we're starting up is pretty horrible!
Better to have a 'startup' bool, or simply set pending_manifests to (size_t)-1 every time you start a plugin after startup, so this never happens?

I did that because it was already used and to avoid yet another member, I'll amend it if it's preferable.